pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -304,3 +398,48 @@ func (m *mgr) reconcileResourceToNamespace(obj runtime.Object, controller string
 		return obj, nil
 	})
 }
+
+func (m *mgr) addRTAnnotation(obj runtime.Object, context string) (runtime.Object, error) {
+	meta, err := meta.Accessor(obj)
+	if err != nil {
+		return obj, err
+	}
+
+	// If the annotation is already there move along
+	if _, ok := meta.GetAnnotations()[roleTemplatesRequired]; ok {
+		return obj, nil
+	}
+
+	rt, err := m.roleTemplateLister.List("", labels.NewSelector())
+	if err != nil {
+		return obj, err
+	}
+
+	annoMap := make(map[string][]string)
+
+	switch context {
+	case "project":
+		for _, role := range rt {
+			if role.Context == context && role.ProjectCreatorDefault {
```

Comment:
Technically, the "context" field is just a hint for the UI on where the RT should be displayed and the lack of a context means it can/should be displayed at BOTH the cluster and project level. So, I do not think you should consider context here (and same for cluster below).

Reply:
Removed